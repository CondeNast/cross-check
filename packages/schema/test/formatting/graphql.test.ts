import { GRAPHQL_SCALAR_MAP, subject, setupSchemaTest, teardownSchemaTest } from "../support";

describe("[schema] formatting - graphql", () => {
  beforeEach(setupSchemaTest);
  afterEach(teardownSchemaTest);

  test("simple", () => {
    let { format } = subject();
    expect(
      format.published.graphql("SimpleArticle", {
        name: "Simple",
        scalarMap: GRAPHQL_SCALAR_MAP
      })
    ).toMatchSnapshot();

    expect(
      format.draft.graphql("SimpleArticle", {
        name: "Simple",
        scalarMap: GRAPHQL_SCALAR_MAP
      })
    ).toMatchSnapshot();
  });

  test("detailed", () => {
    let { format } = subject();
    expect(
      format.published.graphql("MediumArticle", {
        name: "MediumArticle",
        scalarMap: GRAPHQL_SCALAR_MAP
      })
    ).toMatchSnapshot()
  });

  test("relationships", () => {
    let { format } = subject();
    expect(
      format.published.graphql("Related", {
        name: "Related",
        scalarMap: GRAPHQL_SCALAR_MAP
      })
    ).toMatchSnapshot();
  });
});
