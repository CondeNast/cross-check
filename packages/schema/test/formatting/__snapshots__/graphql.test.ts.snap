// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[schema] formatting - graphql detailed 1`] = `
"type MediumArticleAuthor {
  first: SingleLine
  last: SingleLine
}

type MediumArticleGeo {
  lat: Int!
  long: Int!
}

type MediumArticleContributors {
  first: SingleLine
  last: SingleLine
}

type MediumArticle {
  hed: SingleLine!
  dek: String
  body: String!
  author: MediumArticleAuthor
  issueDate: ISODate
  canonicalUrl: Url
  tags: [SingleWord!]
  categories: [SingleLine!]!
  geo: MediumArticleGeo
  contributors: [MediumArticleContributors!]
  relatedArticles: [MediumArticle!]
}"
`;

exports[`[schema] formatting - graphql relationships 1`] = `
"type Related {
  first: SingleLine
  last: String
  person: SimpleArticle!
  articles: [MediumArticle!]
}"
`;

exports[`[schema] formatting - graphql simple 1`] = `
"type Simple {
  hed: SingleLine!
  dek: String
  body: String!
}"
`;

exports[`[schema] formatting - graphql simple 2`] = `
"type Simple {
  hed: String
  dek: String
  body: String
}"
`;
